### PR TITLE
move javascript to the end of body tag.

### PR DESCRIPTION
**Pull request by wuboy0307**
We should move javascript to the end of body tag for better performance

Please see discussion here.
http://stackoverflow.com/questions/1220956/move-jquery-to-the-end-of-body-tag

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://wuboy0307-move-js.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://wuboy0307-move-js.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: frontend/app/views/spree/shared/_foot.html.erb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      if (window.addEventListener)
+          window.addEventListener("load", downloadJSAtOnload, false);
+      else if (window.attachEvent)
+          window.attachEvent("onload", downloadJSAtOnload);
+      else window.onload = downloadJSAtOnload;
+  </script>
+-->
+
+<!-- load page java script after load all library-->
+<% if conten_for?(:foot) %>
```
Comment (JDutil) [inline frontend/app/views/spree/shared/_foot.html.erb]: content_for to fix specs

**Comment by JDutil** [comment]
This has been attempted before and we had to revert it due to some issues with turbolinks not working properly.  Not sure if that has been fixed yet or not, but would be good to do this if we can.

**Comment by wuboy0307** [comment]
What's the detail of the turbolinks issue? I modified this and use them in my project and it works fine. Both in development and production environment.

By the way, this file in spree_i18n should also modified as well.
https://github.com/spree-contrib/spree_i18n/blob/master/app/overrides/spree/layouts/spree_application/add_js_params.html.erb.deface

File: frontend/app/views/spree/shared/_foot.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+          window.addEventListener("load", downloadJSAtOnload, false);
+      else if (window.attachEvent)
+          window.attachEvent("onload", downloadJSAtOnload);
+      else window.onload = downloadJSAtOnload;
+  </script>
+-->
+
+<!-- load page java script after load all library-->
+<% if content_for?(:foot) %>
+    <%= yield :foot %>
```
Comment (jasonfb) [inline frontend/app/views/spree/shared/_foot.html.erb]: this is a good pattern, I use a pattern similar to this as well. However, it means that all inline javascript in Spree and your custom app (which is really a code smell anyway IMHO) needs to be moved into a content_for(:foot) block

I think various parts of spree may be littered still with inline javascript.

**Pull request closed without merging**